[TW-1030] Output display types (#4891)
postmanlabs/postman-docs: src/pages/docs/postman-flows/tutorials/creating-output-displays.md

---

> title: "Creating output displays"

Reviewer: On line `273` in `/docs/postman-flows/reference/blocks-list/`, it looks like we have a link pointing to the outdated URL that uses the outdated page title.

---

<img alt="Output block" src="https://assets.postman.com/postman-labs-docs/creating-charts-tables-and-output/stock-charts.gif"/>

The **Output** block accepts information from other blocks and creates a visual display of that information. You can select from several display types.

For some examples of what you can do with these blocks, check out the [Stock Dashboard](https://www.postman.com/postman/workspace/utility-flows/flow/64123b57c224290033fcb089) and [WordPress Sentiment Analysis](https://www.postman.com/postman/workspace/utility-flows/flow/6413acdb8c4c54003a4ad611) flows.

The image above shows four types of output display types in the [Stock Dashboard](https://www.postman.com/postman/workspace/utility-flows/flow/64123b57c224290033fcb089): a line chart, some text, a gauge, and a boolean. The tutorials below show how to create each of these display types. You can also find links to example flows for other display types in the [Types of output displays with example flows](#types-of-output-displays-with-example-flows) section.

* [Line chart](#line-chart)
* [Text](#text)
* [Gauge](#gauge)

> * [Boolean](#boolean)

Reviewer: Looks great! A few suggestions:

* What do you think about rearranging this a bit to have some text before the GIF? I'm suggesting this because I think it might give users more context about the GIF before they see it.

* This is minor, but what do you think about moving the example flows links to the end of the intro? I'm suggesting this because the links take you away from the LC. My thinking is that this might encourage users to read the entire intro before navigating away from the LC.

* Also, what do you think about adding a "Contents" section? I'm suggesting this because the page has multiple sections, and a "Contents" section might make it easier for users to jump around the page (and give them a quick view of all the sections on the page).

Something similar to:

```
The **Output** block accepts information from other blocks and creates a visual display of that information. You can select from several display types.

<img alt="Output block" src="https://assets.postman.com/postman-labs-docs/creating-charts-tables-and-output/stock-charts.gif"/>

The image above shows four types of output display types: line chart, text, gauge, and boolean. The following tutorials show how to create each of these display types. You can also find links to example flows for other display types in the [Types of output displays with example flows](#types-of-output-displays-with-example-flows) section.

For examples of how to use **Output** blocks in your flows, see the [Stock Dashboard](https://www.postman.com/postman/workspace/utility-flows/flow/64123b57c224290033fcb089) and [WordPress Sentiment Analysis](https://www.postman.com/postman/workspace/utility-flows/flow/6413acdb8c4c54003a4ad611) flows.

## Contents

* [Line chart](#line-chart)
* [Text](#text)
* [Gauge](#gauge)
* [Boolean](#boolean)
* [Types of output displays with example flows](#types-of-output-displays-with-example-flows)
```

## Line chart

> The line chart display type accepts two lists and plots them on x and y axes in a chart. The first list received is the x axis, and the second list is the y axis. To create an example line chart **Output** block:

Reviewer: We mention that the first list is the "x axis" and the second list is the "y axis". But in step 1 it looks like we use "X axis" and "Y axis" keys to define which list applies to which axis. Does the order that each list is received make a difference?

Author: Changed to List 1 and List 2

> The line chart display type accepts two lists and plots them on x and y axes in a chart. The first list received is the x axis, and the second list is the y axis. To create an example line chart **Output** block:

Reviewer: ```suggestion
The **Line Chart** display type accepts two lists and plots them on x and y axes in a chart. The first list received is the x axis, and the second list is the y axis. To create an example line chart **Output** block:
```

Reviewer: Lead-in sentences should be on their own line, and follow the format “To xyz, do the following:”: https://postmanlabs.atlassian.net/wiki/spaces/TW/pages/3978756522/Procedures

1. Create a new flow and enter the following data in the **Start** block:

>     ```bash

Reviewer: Is this JSON in the code block? If so, what do you think about changing `bash` to `json`?

    {
    "X axis": [10, 20, 30, 40, 50, 60, 70],
    "Y axis": [100, 200, 300, 400]
    }
    ```

1. Connect a **Select** block to the **Start** block and select `X axis`.

> 1. Connect another **Select** block to the **Start** block and select `Y axis`.

Reviewer: In this instance, should we format "X axis" and "Y axis" in bold, instead of inline code samples? I'm suggesting this because it seems like these are UI elements that users select, not enter.

1. Connect both **Select** blocks to a single **Output** block.
1. In the **Output** block's dropdown list, select **Line chart** and run the flow. A line chart appears with the data on their respective axes.

## Text

> When an **Output** block receives a text string, it automatically selects the **Text** display type and shows the text. To create an example text **Output** block:

Reviewer: Can the **Output** block also accept a numeric string (`"2"`)? If so, what do you think about replacing "text string" with "string"?

> When an **Output** block receives a text string, it automatically selects the **Text** display type and shows the text. To create an example text **Output** block:

Reviewer: ```suggestion
When an **Output** block receives a text string, it automatically selects the **Text** display type and shows the text.

To create an example text **Output** block, do the following:
```

1. Create a new flow and add a **Send request** block with a GET request to `postman-echo.com/get`.
1. Connect a **Select** block to the **Send request** block and enter `body.headers.host`. This selects a string from the response.
1. Connect an **Output** block to the **Select** block and run the flow. The **Output** block shows the `body.headers.host` string from the response, which in this example is `postman-echo.com`.

## Gauge

> The **Gauge** display type shows a value's position in a range using a semicircular graph. To create an example gauge **Output** block:

Reviewer: ```suggestion
The **Gauge** display type shows a value's position in a range using a semicircular graph.

To create an example gauge **Output** block, do the following:
```

1. Create a new flow and add a **Send request** block with a GET request to `techcrunch.com/wp-json/wp/v2/posts`.

> 1. Connect an **Evaluate** block to the **Send request** block and enter this [FQL](/docs/postman-flows/flows-query-language/introduction-to-fql/): `$count(value1.body)` to count how many items the response body contains.

Reviewer: Are we using "FQL" to also refer to a query written in FQL, or are we calling these "FQL queries"?

1. Connect an **Output**  block to the **Evaluate** block and select **Gauge** from the dropdown list.
1. Run the flow. The gauge shows the number of items in the response body as the shaded portion of the graph.

>   > You can set the minimum and maximum values for the gauge with the **Output** block's settings icon <img alt="Settings icon" src="https://assets.postman.com/postman-docs/icon-settings-v9.jpg#icon" width="16px">.

Reviewer: ```suggestion
  > You can select the **Output** block's settings icon <img alt="Settings icon" src="https://assets.postman.com/postman-docs/icon-settings-v9.jpg#icon" width="16px"> to set the minimum and maximum values for the gauge.
```

## Boolean

> When an **Output** block receives a `True` or `False` message, it automatically selects the **Boolean** display type, showing **True** or **False**. To create an example boolean **Output** block:

Reviewer: ```suggestion
When an **Output** block receives a `True` or `False` message, it automatically selects the **Boolean** display type, showing **True** or **False**.

To create an example boolean **Output** block, do the following:
```

> When an **Output** block receives a `True` or `False` message, it automatically selects the **Boolean** display type, showing **True** or **False**. To create an example boolean **Output** block:

Reviewer: What do you think about formatting True and False as text, instead of formatting as inline code samples? I'm suggesting this because users aren't expected to enter "True" and "False" as values in the block.

Maybe we could rewrite as something similar to: `When an **Output** block receives a message that is either true or false, ...`

1. Create a new flow and add a **Send request** block with a GET request to `postman-echo.com/get`.
1. Connect an **Evaluate** block to the **Send request** block and enter this [FQL](/docs/postman-flows/flows-query-language/introduction-to-fql/): `value1.http.status = 200` to check if the response contains a `200` status code.
1. Connect an **Output**  block to the **Evaluate** block and run the flow. If the response contains a `200` status code, the **Output** block shows a circle that says **True**. If not, the circle shows **False**.

## Types of output displays with example flows

> You can select the output formats listed below from the Output block's dropdown menu. Select the **Example** links to open flows in Postman that demonstrate each output format.

Reviewer: When I see instructions like the second sentence, I assume the docs are referring to instructions in the product, not in the documentation.

To reduce opportunities for confusion, what do you think about removing the second sentence, and making the example link titles more descriptive--such as "`[Bar chart example](https://www.postman.com/postman/workspace/flows-snippets/flow/6411f07120794b0039e76839)`" or "`See an [example flow using a bar chart display type](https://www.postman.com/postman/workspace/flows-snippets/flow/6411f07120794b0039e76839).`".

**Text** - Shows the information connected to the input as lines of text. [Example](https://www.postman.com/postman/workspace/flows-snippets/flow/6414e20b95e5e70033f028c9)

**Number** - Shows a number, if the information connected to the input is a number. [Example](https://www.postman.com/postman/workspace/flows-snippets/flow/6414e8eb95e5e70033f028d4)

**Boolean** - Shows a true or false value. If the value connected to the input isn't true/false or 0/1, it will always display true. [Example](https://www.postman.com/postman/workspace/flows-snippets/flow/6414ebb88c4c54003a4ad797)

> **JSON** - Shows the information connected to the input in JSON format, so you can see and click on hierarchical information. [Example](https://www.postman.com/postman/workspace/flows-snippets/flow/6414ec507221e9003a7a5384)

Reviewer: ```suggestion
**JSON** - Shows the information connected to the input in JSON format, enabling you to view and select data. [Example](https://www.postman.com/postman/workspace/flows-snippets/flow/6414ec507221e9003a7a5384)
```

Reviewer: I'm not confident that "data" is the best word here. My concern is that "hierarchical" increases the readability level.

**Bar Chart** - Shows one or more sets of information connected to the input. [Example](https://www.postman.com/postman/workspace/flows-snippets/flow/6411f07120794b0039e76839)

**Line Chart** - Shows a line for X and Y coordinates connected to the input. [Example](https://www.postman.com/postman/workspace/flows-snippets/flow/641242fd8c4c54003a4ad507)

> **Gauge** - Shows a number; generally for when the input is a percent or out of one hundred. [Example](https://www.postman.com/postman/workspace/flows-snippets/flow/64124134c224290033fcb08b)

Reviewer: ```suggestion
**Gauge** - Shows a number when the input is a percent or out of 100. [Example](https://www.postman.com/postman/workspace/flows-snippets/flow/64124134c224290033fcb08b)
```

Reviewer: Microsoft Style Guide suggests using numerals for 10 or greater: https://learn.microsoft.com/en-us/style-guide/numbers

**Table** - Shows the connected input as a table with rows and columns. [Example](https://www.postman.com/postman/workspace/flows-snippets/flow/64124607c224290033fcb092)

> **Markdown** - Shows the text with all of the markdown annotations displayed. [Example](https://www.postman.com/postman/workspace/flows-snippets/flow/641249577221e9003a7a5107)

Reviewer: ```suggestion
**Markdown** - Shows the text with all of the Markdown annotations displayed. [Example](https://www.postman.com/postman/workspace/flows-snippets/flow/641249577221e9003a7a5107)
```

> **URL** - Shows a hyperlink to the URL that is connected to the input. [Example](https://www.postman.com/postman/workspace/flows-snippets/flow/64124dc8c224290033fcb098)

Reviewer: ```suggestion
**URL** - Shows a hyperlink to the URL that's connected to the input. [Example](https://www.postman.com/postman/workspace/flows-snippets/flow/64124dc8c224290033fcb098)
```

> **Image** - Shows an image from a URL that is connected to the input. [Example](https://www.postman.com/postman/workspace/flows-snippets/flow/64124e997221e9003a7a510c)

Reviewer: ```suggestion
**Image** - Shows an image from a URL that's connected to the input. [Example](https://www.postman.com/postman/workspace/flows-snippets/flow/64124e997221e9003a7a510c)
```

> **Video** - Shows a video from a URL that is connected to the input and enables it to be played. [Example](https://www.postman.com/postman/workspace/flows-snippets/flow/64124f65c224290033fcb09b)

Reviewer: ```suggestion
**Video** - Shows a video from a URL that's connected to the input and enables it to be played. [Example](https://www.postman.com/postman/workspace/flows-snippets/flow/64124f65c224290033fcb09b)
```

> **YouTube** - Shows a YouTube video from a URL that is connected to the input and enables it to be played. [Example](https://www.postman.com/postman/workspace/flows-snippets/flow/6414ef7bc224290033fcb310)

Reviewer: ```suggestion
**YouTube** - Shows a YouTube video from a URL that's connected to the input and enables it to be played. [Example](https://www.postman.com/postman/workspace/flows-snippets/flow/6414ef7bc224290033fcb310)
```
